hls4ml/model/optimizer/__init__.py

```diff
@@ -41,6 +41,7 @@
 register_pass('optimize_pointwise_conv', OptimizePointwiseConv)
 register_pass('clone_output', CloneOutput)
 register_pass('reshape_stream', ReshapeStream)
+register_pass('remove_final_reshape', RemoveFinalReshape)
```

Comment:
When using `io_stream`, the `Reshape` layer is replaced with a `Repack` so the `RemoveFinalReshape` doesn't pick it up. The model still works, but that final reshape is redundant and costs some unnecessary resources, as in the `Flatten` case. 

If we swap the order of these lines to run `RemoveFinalReshape` before `ReshapeStream`, then we can additionally remove `Reshape` as the final layer for `io_stream` and save some resources.

Reply:
Swapped, just in case we want to remove it in the future for `io_stream`, but currently we are keeping it for `io_stream`